vnext/Microsoft.ReactNative/Modules/LinkingManagerModule.cpp

```diff
@@ -3,76 +3,100 @@
 
 #include "pch.h"
 
-#include <Utils/ValueUtils.h>
 #include <winrt/Windows.System.h>
+#include "JSValueWriter.h"
+#include "LinkingHelper.h"
 #include "LinkingManagerModule.h"
-#include "Unicode.h"
-
-#include <cxxreact/Instance.h>
-#include <cxxreact/JsArgumentHelpers.h>
 
 #if _MSC_VER <= 1913
 // VC 19 (2015-2017.6) cannot optimize co_await/cppwinrt usage
 #pragma optimize("", off)
 #endif
 
-using Callback = facebook::xplat::module::CxxModule::Callback;
-
 using namespace winrt;
 using namespace Windows::Foundation;
 using namespace Windows::System;
-using namespace ::Microsoft::Common::Unicode;
 
 namespace Microsoft::ReactNative {
 
 //
 // LinkingManagerModule helpers
 //
 
-static fire_and_forget openURLAsync(Uri uri, Callback success, Callback error) {
+static fire_and_forget openURLAsync(Uri uri, React::ReactPromise<React::JSValue> result) {
   if (co_await Launcher::LaunchUriAsync(uri)) {
-    success({true});
+    result.Resolve({true});
   } else {
-    error({folly::dynamic::object("code", 1)("message", "Unable to open URL: " + Utf16ToUtf8(uri.DisplayUri()))});
+    const auto message = L"Unable to open URL: " + uri.DisplayUri();
+    result.Reject(message.c_str());
   }
 }
 
-static fire_and_forget canOpenURLAsync(Uri uri, Callback success, Callback /*error*/) {
+static fire_and_forget canOpenURLAsync(Uri uri, React::ReactPromise<React::JSValue> result) {
   auto status = co_await Launcher::QueryUriSupportAsync(uri, LaunchQuerySupportType::Uri);
   if (status == LaunchQuerySupportStatus::Available) {
-    success({true});
+    result.Resolve({true});
   } else {
-    success({false});
+    result.Resolve({false});
   }
 }
 
+static std::mutex s_mutex;
+static std::vector<LinkingManager *> s_linkingModules;
```

Comment:
why are these not a class static anymore?

Reply:
ah I see your comment on the header. My personal take is that making them members of the class makes it a lot more obvious when looking at the debugger output about who is using the static, otherwise it's just another global in the m.rn namespace